onecgiar-pr-server/src/shared/microservices/email-notification-management/email-notification-management.service.spec.ts

```diff
@@ -0,0 +1,202 @@
+      const logSpy = jest.spyOn(service['_logger'], 'log');
+      await service.onModuleInit();
+      expect(logSpy).toHaveBeenCalledWith(
+        'Successfully connected to RabbitMQ Email MicroService',
+      );
+    });
+
+    it('should log error message on failed connection', async () => {
+      const errorSpy = jest.spyOn(service['_logger'], 'error');
+      jest
+        .spyOn(clientProxy, 'connect')
+        .mockRejectedValue(new Error('Connection failed'));
+      await service.onModuleInit();
+      expect(errorSpy).toHaveBeenCalledWith(
+        'Failed to connect to RabbitMQ Email MicroService',
+        'Connection failed',
+      );
+    });
+  });
+
+  describe('sendEmail', () => {
+    it('should emit the email with correct payload', () => {
+      const email: ConfigMessageDto = {
+        from: { email: 'support@prms.pr', name: '[PRMS]' },
+        emailBody: {
+          subject: 'Test email',
+          to: [userTest.email],
+          cc: [userTest.email],
+          message: {
+            text: 'Contribution request',
+          },
+        },
+      };
+      const expectedPayload = {
+        auth: {
+          username: env.MS_NOTIFICATION_USER,
+          password: env.MS_NOTIFICATION_PASSWORD,
+        },
+        data: email,
+      };
+      service.sendEmail(email);
+      expect(clientProxy.emit).toHaveBeenCalledWith('send', expectedPayload);
+    });
+  });
+
+  describe('buildEmailData', () => {
+    it('should build email data for email_template_contribution with correct label', () => {
+      const data: BuildEmailDataDto = {
+        initOwner: {
+          id: 1,
+          official_code: 'INIT-01',
+          name: 'Owner Initiative',
+          short_name: 'Owner',
+        },
+        initContributing: {
+          id: 2,
+          name: 'Contributing Initiative',
+          official_code: 'INIT-02',
+          short_name: 'Contributing',
+        },
+        user: {
+          id: 1,
+          first_name: 'John',
+          last_name: 'Doe',
+          email: 'jhon@doe.com',
+        },
+        result: {
+          result_code: 1,
+          title: 'Sample Result',
+          version_id: 1,
+        },
+      };
+
+      process.env.IS_PRODUCTION = 'false';
+      let result = service.buildEmailData(EmailTemplate.CONTRIBUTION, data);
+
+      expect(result).toEqual({
+        cc: ['jhon@doe.com'],
+        subject:
+          '[PRMS Testing] INIT-01 requests INIT-02 to contribute to Result 1 - ',
+        initContributingName: 'Contributing',
+        requesterName: 'John Doe',
+        initContributing: 'INIT-02 Contributing',
+        initOwner: 'INIT-01 Owner',
+        resultUrl: `${env.RESULTS_URL}1/general-information?phase=1`,
+        result: '1 - Sample Result',
+        resultNotificationUrl: `${env.NOTIFICATION_MODULE_URL}requests/received?phase=1&init=2&search=John Doe from INIT-01 has requested inclusion of INIT-02 as a contributor to result 1 - Sample Result`,
```

Comment:
@JuanGuzman-io shouldn't spaces on URLs be escaped? replaced by `%20` I mean

Reply:
@xKeCo If I replace the spaces, will the search in the notification be affected?